pkg/manifest/yaml.go

```diff
@@ -158,6 +159,75 @@ func (v *Service) SetName(name string) error {
 	return nil
 }
 
+func (a *Annotations) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var rawAnnotations []interface{}
+	if err := unmarshal(&rawAnnotations); err != nil {
+		return fmt.Errorf("failed to unmarshal annotations: %w", err)
+	}
+
+	var annotations []Annotation
+
+	for _, raw := range rawAnnotations {
+		switch item := raw.(type) {
+		case string:
+			parts := strings.SplitN(item, "=", 2)
+			if len(parts) != 2 {
+				return fmt.Errorf("invalid annotation format: %s", item)
+			}
+			annotations = append(annotations, Annotation{Key: parts[0], Value: parts[1]})
+
+		case map[interface{}]interface{}:
+			// Handle annotations as map[interface{}]interface{}
+			for k, v := range item {
+				keyStr, ok := k.(string)
+				if !ok {
+					return fmt.Errorf("annotation key is not a string: %v", k)
+				}
+
+				// Try to parse the value as JSON
+				var jsonValue interface{}
+				if jsonStr, ok := v.(string); ok {
+					if err := json.Unmarshal([]byte(jsonStr), &jsonValue); err == nil {
+						// Successfully unmarshaled JSON string into a map
+						annotations = append(annotations, Annotation{Key: keyStr, Value: jsonValue})
+					} else {
+						// If not a valid JSON, store the string as-is
+						annotations = append(annotations, Annotation{Key: keyStr, Value: jsonStr})
+					}
+				} else {
+					// Store as is if not a JSON string
+					annotations = append(annotations, Annotation{Key: keyStr, Value: v})
+				}
```

Comment:
why json unmarshal here?

Reply:
If the nested json needs to be formatted, otherwise we add the value as it is.